packages/core/src/labels.ts

```diff
@@ -16,7 +18,13 @@ export type Labels = Record<string, string | number | boolean>;
  * entirely.
  */
 export function withLabels<T>(operation: Operation<T>, labels: Labels): Operation<T> {
-  if(operation) {
+  if (isObjectOperation<T>(operation)) {
+    let original = operation[Symbol.operation];
+    Object.assign(operation, {
+      ...labels,
+      [Symbol.operation]: original
+    });
```

Comment:
Why mutate if we don't have to?

```suggestion
    return {
      ...labels,
      [Symbol.operation]: original
    };
```

Reply:
Because that's how `withLabels` works according the documentation `Apply the given labels to an operation.` So it doesn't create new operation. Or I don't understand something? Originally `withLabels` replaces some labels in operation, so I tried to keep this logic. And if we change it, we will not replace labels, instead we create another operation, keep old labels and start depending on `controller` implementation.

Reply:
So real replacement will happened in `controller`

Reply:
I mean this
![image](https://user-images.githubusercontent.com/6397708/149151457-257b905c-2203-470b-b90b-633a5e254481.png)

Reply:
Should we change tests?

Reply:
You are right. Even though this might be better, it is a breaking API change and cannot be added to 2.0

`withLabels()` must have the exact same behavior